src/app/shared/components/search-bar/search-bar.component.ts

```diff
@@ -1,133 +1,69 @@
-import { Component, inject, OnInit } from '@angular/core';
-import { FilterService, SelectItemGroup } from 'primeng/api';
+import { Component, inject, Input, OnInit } from '@angular/core';
+import { Job } from '../../../features/jobs/models/job';
+import { JobService } from '../../../features/jobs/service/job.service';
+import { Observable } from 'rxjs';
+import { Router } from '@angular/router';
 
-interface AutoCompleteCompleteEvent {
-  originalEvent: Event;
-  query: string;
-}
 @Component({
   selector: 'app-search-bar',
   templateUrl: './search-bar.component.html',
   styleUrl: './search-bar.component.scss',
 })
 export class SearchBarComponent implements OnInit {
-  selectedCity: any;
-  filteredGroups: any[] = [];
-  groupedCities: SelectItemGroup[] | undefined;
-
-  private readonly filterService: FilterService = inject(FilterService);
+  private readonly jobService: JobService = inject(JobService);
+  private readonly router = inject(Router);
+  @Input() jobList?: Observable<Job[]>;
```

Comment:
Do not put an question mark after jobList input, do a jobList$!

Remember, put the $ also because it is an observable and makes it easier to understand